fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadTaskScheduler.java

```diff
@@ -219,6 +219,17 @@ private void scheduleOneTask(RoutineLoadTaskInfo routineLoadTaskInfo) throws Exc
             return;
         }
 
+        // Double check if task has been abandoned
+        if (!routineLoadManager.checkTaskInJob(routineLoadTaskInfo.getId())) {
+            releaseBeSlot(routineLoadTaskInfo);
+            // task has been abandoned while renew task has been added in queue
```

Comment:
If the task is abandoned after this check, will there still be the same problem?

Reply:
BE slot will be released in [ClearTasks()](https://github.com/StarRocks/starrocks/blob/main/fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadJob.java#L1336), it's OK.